### DISCUSSION TITLE
Prevent disabled days from being included in keyboard events

### DISCUSSION THREAD

**Discussion by leighsn** [gpbl/react-day-picker#1095] [Support]
**Is your feature request related to a problem? Please describe.**
Thanks for providing this neat calendar. I'm on v7.4.0 One issue: I would expect disabled days to be omitted from the default keyboard navigation order so that keyup and keydown events would focus the next / previous available day.

**Describe the solution you'd like**
A simple and general solution might be to provide a way to pass props into the Day component, perhaps by adding a key to modifier props. This would enable users to overwrite default properties on the Day component.

**Describe alternatives you've considered**
I tried to implement this myself using the onDayKeyDown handler:

```jsx
	onDayKeyDown = (day, m, e) => {
		const { fromDate, toDate } = this.state;
		const { key } = e
		if (key === 'ArrowDown' || key === 'ArrowRight' && day !== toDate) {
			this.focusNextDay(day);
			return;
		}
		if (key === 'ArrowUp' || key === 'ArrowDown' && day !== fromDate) {
			this.focusPreviousDay(day);
			return;
		}
	}
```
The issue I ran into is that the day that is being passed into this handler is not the day that was in focus during the key event - it's the day in focus after the key event navigates the user to the next / previous / up/ down day. 

**Additional context**
Thanks for considering.

**Comment by gpbl** [comment]
> I would expect disabled days to be omitted from the default keyboard navigation order so that keyup and keydown events would focus the next / previous available day.

Yes I'd expect that too! Defenitely a bug / feature.

> The issue I ran into is that the day that is being passed into this handler is not the day that was in focus during the key event 

Do you know if this is expected or not? We should investigate, otherwise it looks like a serious bug.

I agree – fixing the problem by skipping the disabled day would work.

**Comment by joshwreford** [comment]
Hey @gpbl. Was this issue resolved? We're having this problem in our app. We're currently using react-day-picker v7.3.2. Thanks.
